src/app/(pages)/query/components/searchForm/searchForm.test.tsx

```diff
@@ -186,4 +206,52 @@ describe("SearchForm", () => {
     expect(zip).not.toHaveValue(badAddressZip);
     expect(zip).toHaveValue("");
   });
+
+  it("shows and toggles patient match checkbox", async () => {
+    const queryParams = "";
+    (useSearchParams as jest.Mock).mockReturnValue(
+      new URLSearchParams(queryParams),
+    );
+    (getFhirServerConfigs as jest.Mock).mockResolvedValue([
+      {
+        name: "Matching Server",
+        patientMatchConfiguration: {
+          enabled: true,
+          supportsMatch: true,
+          onlyCertainMatches: false,
+          onlySingleMatch: false,
+          matchCount: 5,
+        },
+      },
+    ]);
+
+    const { user } = renderWithUser(
+      <RootProviderMock currentPage="/query">
+        <SearchForm
+          setMode={jest.fn()}
+          setLoading={jest.fn()}
+          setPatientDiscoveryQueryResponse={jest.fn()}
+          selectedFhirServer="Matching Server"
+          setFhirServer={jest.fn()}
+          fhirServers={["Matching Server"]}
+          setUncertainMatchError={jest.fn()}
+        />
+      </RootProviderMock>,
+    );
+
+    const advancedButton = await screen.findByRole("button", {
+      name: "Advanced",
+    });
+    await user.click(advancedButton);
+
+    const checkbox = await screen.findByRole("checkbox", {
+      name: /enable patient match/i,
+    });
+
+    expect(checkbox).toBeVisible();
+    expect(checkbox).toBeChecked();
+
+    await user.click(checkbox);
+    expect(checkbox).not.toBeChecked();
+  });
```

Comment:
would love to also have a test that checks this interaction formats the FHIR request in the right way. Any chance we could put a spy (or maybe on fetch itself, we're doing something similar [here](https://github.com/CDCgov/dibbs-query-connector/pull/765/files#diff-4ae607a1cb9239151fa9cd0bc96d2fb77c9c2b2f65f532bed3651700240400ebR17)) onto the relevant function and make sure the FHIR request has the relevant `$match` shape? 

Feel free to do this in a followup: this PR definitely is good as is.